src/Microsoft.Azure.WebJobs.Host/Executors/TriggeredFunctionExecutor.cs

```diff
@@ -35,7 +35,8 @@ public FunctionDescriptor Function
             var context = new FunctionInstanceFactoryContext<TTriggerValue>()
             {
                 TriggerValue = (TTriggerValue)input.TriggerValue,
-                ParentId = input.ParentId
+                ParentId = input.ParentId,
+                Id = input.Id.GetValueOrDefault(Guid.NewGuid())
```

Comment:
Being on the invocation path, please use `input.Id ?? Guid.NewGuid()` for efficiency.

Reply:
Oh cool, I didn't know that ?? was faster.

Reply:
It's not just ??. With the current implementation, you're always creating a NewGuid, instead of allowing this to short circuit in the scenarios where the id is already set, which the recommended change would do.